azure-iot-hub/tests/test_iothub_amqp_client.py

```diff
@@ -34,58 +23,109 @@
     "expiryTimeUtc": 1584727659,
     "messageId": 42,
 }
-fake_empty_prop = {}
-"""----Shared fixtures----"""
-
+fake_hostname = "hostname.mytest.net"
+fake_shared_access_key_name = "test_key_name"
+fake_shared_access_key = "Zm9vYmFy"
+fake_token_scope = "https://token.net/.default"
+fake_token = "fasdjkhg"
+fake_token_expiry = 1584727659
 
-@pytest.fixture(scope="function", autouse=True)
-def mock_uamqp_SendClient(mocker):
-    mock_uamqp_SendClient = mocker.patch.object(uamqp, "SendClient")
-    return mock_uamqp_SendClient
 
+class SharedAmqpClientTests(object):
+    @pytest.fixture(autouse=True)
+    def mock_uamqp_SendClient(self, mocker):
+        mock_uamqp_SendClient = mocker.patch.object(uamqp, "SendClient")
+        return mock_uamqp_SendClient
 
-@pytest.mark.describe("IoTHubAmqpClient - Amqp Client Connections")
-class TestIoTHubAmqpClient(object):
-    @pytest.mark.it("Send Message To Device")
-    def test_send_message_to_device(self, mocker, mock_uamqp_SendClient):
-        iothub_amqp_client = IoTHubAmqpClient(
-            fake_hostname, fake_shared_access_key_name, fake_shared_access_key
-        )
-        iothub_amqp_client.send_message_to_device(fake_device_id, fake_message, fake_app_prop)
+    @pytest.mark.it("Sends messages with application properties using the uamqp SendClient")
+    def test_send_message_to_device_app_prop(self, client, mock_uamqp_SendClient):
+        client.send_message_to_device(fake_device_id, fake_message, fake_app_prop)
         amqp_client_obj = mock_uamqp_SendClient.return_value
 
         assert amqp_client_obj.queue_message.call_count == 1
         assert amqp_client_obj.send_all_messages.call_count == 1
 
-    @pytest.mark.it("Send Message To Device system prop")
-    def test_send_message_to_device_sys_props(self, mocker, mock_uamqp_SendClient):
-        iothub_amqp_client = IoTHubAmqpClient(
-            fake_hostname, fake_shared_access_key_name, fake_shared_access_key
-        )
-        iothub_amqp_client.send_message_to_device(fake_device_id, fake_message, fake_sys_prop)
+    @pytest.mark.it("Sends messages with system properties using the uamqp SendClient")
```

Comment:
As above, there are also some additional assertions you should be making here. But also you probably should have another test case for messages that don't have any properties provided (i.e. `{}`), since that's optional.